### ISSUE TITLE
Configurable status checks for automerge

### ISSUE THREAD

**Issue by rarkins** [singapore/renovate#359]
Applies to GitHub, maybe GitLab.

Some status checks may be "optional", so we should let the user specify if they want us to check for only some.

**Comment by rarkins** [comment]
Cc @adam-moss this is kind of related to #1586. I think people could use this (unimplemented) feature to specify a custom set of status checks to look for. But nobody seems to have +1'd this in 9 months so it will sit as low priority

**Comment by adam-moss** [comment]
Is this related to GitHub's concept of status checks on protected branches?  If so GitLab doesn't have that concept currently.

If it more generic - as in individual jobs within a CI suite - it should be relatively simple if they're broken out as individual jobs.  If it is to use the defaults that something like travis does, where it injects your install & test steps it could be considerably more challenging.

**Comment by rarkins** [comment]
Yeah, I guess this is just a GitHub thing then if GitLab doesn't have the equivalent concept.

**Comment by rarkins** [comment]
I'm not going to implement this. But first need to rename `requiredStatusChecks=null` to `ignoreTests=true`.

**Comment by rarkins** [comment]
#11163

**Comment by rarkins** [comment]
Reserving this issue again for the concept of people being able to configure either:
- mandatory status checks (in case platform is not enforcing them) so that Renovate doesn't automerge too soon, or
- status checks to ignore (so that Renovate doesn't skip automerging unnecessarily)

We may want to first refactor status checks handling to a common worker instead of reimplementing in each platform like today.

**Comment by lmilbaum** [comment]
Is it still relevant for implementation? I am still learning this project so I can't promise anything.

**Comment by rarkins** [comment]
@lmilbaum we haven't seen a lot of demand for this, and each of the platforms has changed/improved their automerge support over time, so I think this feature doesn't have a high "impact vs work" ratio. We'd love to have you as a contributor - would you like us to suggest any features? Also FYI we have a Slack channel dedicated for contributors - please email me if you'd like an invite

**Comment by lmilbaum** [comment]
> @lmilbaum we haven't seen a lot of demand for this, and each of the platforms has changed/improved their automerge support over time, so I think this feature doesn't have a high "impact vs work" ratio. We'd love to have you as a contributor - would you like us to suggest any features? Also FYI we have a Slack channel dedicated for contributors - please email me if you'd like an invite

email sent. Thanks.
